go.mod

```diff
@@ -3,95 +3,78 @@ module github.com/openshift/route-monitor-operator
 go 1.21
 
 require (
-	github.com/go-logr/logr v1.2.3
+	github.com/go-logr/logr v1.4.1
 	github.com/google/gofuzz v1.2.0
 	github.com/hashicorp/go-version v1.6.0
 	github.com/onsi/ginkgo v1.16.5
-	github.com/onsi/gomega v1.27.4
-	github.com/openshift/api v0.0.0-20230213134911-7ba313770556
-	github.com/openshift/hypershift v0.0.0-20230320142408-f88fd0deab44
+	github.com/onsi/gomega v1.30.0
+	github.com/openshift/api v0.0.0-20240214165302-89248c87b7fc
+	github.com/openshift/hypershift/api v0.0.0-20240401231845-020ef717e96f
 	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.63.0
-	github.com/prometheus/common v0.42.0
+	github.com/prometheus/common v0.45.0
 	github.com/rhobs/obo-prometheus-operator/pkg/apis/monitoring v0.60.0-rhobs1
 	go.uber.org/mock v0.4.0
 	gopkg.in/inf.v0 v0.9.1
-	k8s.io/api v0.26.3
-	k8s.io/apiextensions-apiserver v0.26.3
-	k8s.io/apimachinery v0.26.3
-	k8s.io/client-go v12.0.0+incompatible
-	k8s.io/kube-openapi v0.0.0-20230308215209-15aac26d736a
-	sigs.k8s.io/controller-runtime v0.14.4
+	k8s.io/api v0.29.2
+	k8s.io/apiextensions-apiserver v0.29.2
+	k8s.io/apimachinery v0.29.2
+	k8s.io/client-go v0.29.2
+	k8s.io/kube-openapi v0.0.0-20231010175941-2dd684a91f00
+	sigs.k8s.io/controller-runtime v0.17.2
```

Comment:
:raised_hands:

Reply:
Great question - controller-runtime `v0.18.0` is for K8s `v1.30.z` (and we aren't quite ready for it yet, that's OCP 4.17) https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.18.0